libraries/AP_InertialSensor/AP_InertialSensor_RST.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+
+#include <AP_HAL/I2CDevice.h>
+#include <AP_HAL/SPIDevice.h>
+#include <Filter/Filter.h>
+#include <Filter/LowPassFilter2p.h>
+
+#include "AP_InertialSensor.h"
+#include "AP_InertialSensor_Backend.h"
+
+class AP_InertialSensor_RST : public AP_InertialSensor_Backend
+{
+public:
+    AP_InertialSensor_RST(AP_InertialSensor &imu,
+                              AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_gyro,
+                              AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_accel,
+                              enum Rotation rotation_a,
+                              enum Rotation rotation_g);
+
+    virtual ~AP_InertialSensor_RST();
+
+    // probe the sensor on SPI bus
+    static AP_InertialSensor_Backend *probe(AP_InertialSensor &imu,
+                                              AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_gyro,
+                                              AP_HAL::OwnPtr<AP_HAL::SPIDevice> dev_accel,
+                                              enum Rotation rotation_a,
+                                              enum Rotation rotation_g);
+
+    /* update accel and gyro state */
+    bool update() override;
+
+    void start(void) override;
+
+private:
+    bool _init_sensor();
+    bool _init_gyro();
+    bool _init_accel();
+    void gyro_measure();
+    void accel_measure();
+
+    AP_HAL::OwnPtr<AP_HAL::SPIDevice> _dev_gyro;//i3g4250d
+    AP_HAL::OwnPtr<AP_HAL::SPIDevice> _dev_accel;//iis328dq
+    AP_HAL::Semaphore *_gyro_spi_sem;
+    AP_HAL::Semaphore *_accel_spi_sem;
+
+    // gyro whoami
+    uint8_t _gyro_whoami;
+    // accel whoami
+    uint8_t _accel_whoami;
```

Comment:
like the semaphores, these are only used in one function and don't need to be in the object. Please make them local.